packages/common/src/location/location.ts

```diff
@@ -68,7 +68,7 @@ export class Location implements OnDestroy {
   constructor(locationStrategy: LocationStrategy) {
     this._locationStrategy = locationStrategy;
     const browserBaseHref = this._locationStrategy.getBaseHref();
-    this._baseHref = stripTrailingSlash(_stripIndexHtml(browserBaseHref));
+    this._baseHref = _stripOrigin(stripTrailingSlash(_stripIndexHtml(browserBaseHref)));
```

Comment:
```suggestion
    const baseHref = this._locationStrategy.getBaseHref();
    this._basePath = _stripOrigin(stripTrailingSlash(_stripIndexHtml(baseHref)));
```

Also a nit, but what do you think about updating some of these variable names? `browserBaseHref` is a bit misleading because it makes me think it's the `base href` from the browser, but it could be from APP_BASE_HREF. Also `this._baseHref` is misleading because it's _not_ the same as `baseHref` because we only want to keep the `path` part due to how it's used in this class.